src/www/containers/userprofile/UserProfile.js

```diff
@@ -0,0 +1,260 @@
+          shown: 3,
+        },
+        actionsOrganized: {
+          list: actionsOrganized,
+          shown: 3,
+        },
+        edcountitMode: "off",
+        teamsShown: 3,
+      };
+
+      showCards = (cards, count) => {
+        const { t } = this.props;
+        return Object.keys(cards).map((key, index) => {
+          const card = cards[key];
+          if (index >= count) {
+            return null;
+          }
+          return (
+            <li>
+              <span className="Card">
+                <img className="Card-image" src={card.photo} alt="" />
+                <div className="Card-body">
+                  <h4>{card.name}</h4>
+                  <p>{card.description}</p>
+                  <div className="Card-categories">
+                    <h4>{t("profile.categories")}:</h4>
+                    <ul>
+                      {Object.keys(card.categories).map((cKey) => {
+                        const category = card.categories[cKey];
+                        return <li>{category.name}</li>;
+                      })}
+                    </ul>
+                  </div>
+                </div>
+              </span>
+            </li>
+          );
+        });
+      };
+
+      expandActions = (actions) => {
+        const { t } = this.props;
+        const actionsShown = this.state[actions].shown;
+        const available = Object.keys(this.state[actions].list).length;
+
+        if (available <= minCardsShown) {
+          return null;
+        } else if (actionsShown > minCardsShown) {
+          return (
+            <p
+              onClick={() =>
+                this.setState({
+                  [actions]: { ...this.state[actions], shown: minCardsShown },
+                })
+              }
+            >
+              {t("profile.show-less")}
+            </p>
+          );
+        } else {
+          return (
+            <p
+              onClick={() =>
+                this.setState({
+                  [actions]: { ...this.state[actions], shown: available },
+                })
+              }
+            >
+              {t("profile.show-more")}
+            </p>
+          );
+        }
+      };
+
+      showCategories = (categories) => {
+        return Object.keys(categories).map((key) => {
+          const category = categories[key];
+          return (
+            <li>
+              <div className="Category">
+                <img className="Category-image" src={category.photo} alt="" />
+                <p>{category.name}</p>
+              </div>
+            </li>
+          );
+        });
+      };
+
+      showBio = (initialBio) => {
+        const { user, t } = this.props;
+        if (this.state.editMode) {
+          return (
+            <div>
+              {/*TODO: should send request to server */}
+              <textarea
+                className="Edit-bio-input"
+                defaultValue={initialBio}
+                onChange={(e) => (user.bio = e.target.value)}
```

Comment:
Let's trim this as well.
```suggestion
                onChange={(e) => { user.bio = e.target.value.trim() }}
```